### ISSUE TITLE
redesign get_state

### ISSUE THREAD

**Issue by vincentropy** [arenarium/battleground_core#108]
There is a need for multiple views on the game state. For example, in games where one player only has a limited vision. We should design a solution for this.

**Comment by phylyc** [comment]
check the perception mod in the arena.mods folder. ;)

**Comment by vincentropy** [comment]
Yes, that's exactly what I was looking for :D

I might want to change the signature just a little. How about a signature like this:
``` python
def get_state(self, observer=None):
""" 
Get complete game state by default.
  observer: Pass an integer index to get the game state as viewed by an individual player.
"""
# ... logic ...
return game_state
```
 
that way we can also  give updated gamestates in 3+ player games to all players even when it is not their turn. This may also be interesting for people observing the game.

**Comment by phylyc** [comment]
Yes, we can do that and I'd also prefer that. I was returning the player pov state by default to not having to change anything in game_runner.run_game. But if we are ok to change 
`engine_state = self.game_engine.get_state()`
into 
`engine_state = self.game_engine.get_state(player_index)`
then we can also get rid of the whole get_save_state story and get the state to save through
`state = self.game_engine.get_state()`

This has as a side effect that all get_state methods of all games have to have the observer_id input at least defined, even if the returned object is completely independent of it.
